### PR TITLE
Refactoring SPIP Modules for Windows Compatibility and Incorporating SPIP Mixin

### PR DESCRIPTION
**Pull request by Chocapikk**
**Hello Metasploit Team,**

I am currently working on refactoring the SPIP modules to ensure they are compatible with Windows and also adapting them to the new module syntax. The goal is to enhance integration and improve cross-platform compatibility.

Modules: 
- `exploit/multi/http/spip_rce_form` (Moved from `exploit/unix/webapp`)
- `exploit/unix/webapp/spip_connect_exec` (Moved from `exploit/unix/webapp`)
- `exploit/multi/http/spip_porte_plume_previsu_rce` (Missing the mixin only)

Additionally, I would greatly appreciate it if @jvoisin's [pull request regarding the SPIP mixin](https://github.com/rapid7/metasploit-framework/pull/19401) could be merged, as it would allow us to refactor all SPIP modules to use the mixin consistently. 

For now, I'm marking this PR as a draft because I'm still actively working on these changes.

### CODE REVIEW

File: modules/exploits/multi/http/spip_rce_form.rb [hunk 1]
```diff
@@ -33,49 +34,43 @@ def initialize(info = {})
           [ 'CVE', '2023-27372' ],
         ],
         'Privileged' => false,
-        'Platform' => %w[php linux unix],
+        'Platform' => ['php', 'unix', 'linux', 'win'],
```
Comment (jvoisin) [inline modules/exploits/multi/http/spip_rce_form.rb:37]:
```suggestion
        'Platform' => %w[php, unix, linux, win],
```

File: modules/exploits/multi/http/spip_rce_form.rb [hunk 2]
```diff
@@ -33,49 +34,43 @@ def initialize(info = {})
...
         'DefaultTarget' => 0,
         'DisclosureDate' => '2023-02-27'
       )
     )
-
-    register_options(
-      [
-        OptString.new('TARGETURI', [true, 'The base path to SPIP application', '/']),
-      ]
-    )
```
Comment (jvoisin) [inline modules/exploits/multi/http/spip_rce_form.rb:78]: Why this removal?

File: modules/exploits/multi/http/spip_rce_form.rb [hunk 3]
```diff
@@ -149,16 +155,13 @@ def exploit
...
 
-    oubli = ''
-    case target['Type']
-    when :php_memory
-      oubli = "s:#{payload.encoded.length + 6 + 2}:\"<?php #{payload.encoded}?>\";"
-    when :unix_memory
-      oubli = "s:#{payload.encoded.length + 14 + 4}:\"<?php system('#{payload.encoded}')?>\";"
-    end
+    phped_payload = target['Arch'] == ARCH_PHP ? payload.encoded : php_exec_cmd(payload.encoded)
+    b64_payload = framework.encoders.create('php/base64').encode(phped_payload)
```
Comment (smcintyre-r7) [inline modules/exploits/multi/http/spip_rce_form.rb:161]: `#php_exec_cmd` is also doing base64 encoding. Is it necessary to double base64 encode it or can that part be removed so it's only the `system()` call to execute the command.

File: modules/exploits/multi/http/spip_rce_form.rb [hunk 4]
```diff
@@ -136,6 +131,17 @@ def execute_command(cmd, args = {})
     )
   end
 
+  def php_exec_cmd(encoded_payload)
+    dis = '$' + Rex::Text.rand_text_alpha(rand(4..7))
```
Comment (smcintyre-r7) [inline modules/exploits/multi/http/spip_rce_form.rb:135]: We don't currently have a PHP generator for this sort of thing yet, but would you mind using the default Rex::RandomIdentifier? https://github.com/rapid7/rex-random_identifier/blob/master/lib/rex/random_identifier/generator.rb#L23C7-L23C39

The idea is to ensure that we're not accidentally creating variable names that are invalid in the context of PHP. We don't want to block this on waiting for a PHP definition, but we should add one in the future.

**Comment by dwelch-r7** [comment]
I've landed #19401 if you'd like to rebase this PR

**Comment by Chocapikk** [comment]
Hi everyone,

I've run into some issues with the commit history in this PR, and I'm not sure how to resolve it cleanly. To avoid any confusion and to ensure the changes are properly submitted, I'm going to close this PR and open a new one with a clean history.

Thanks for your understanding, and sorry for the inconvenience!

**Pull request closed without merging**